src/game/Strategic/Strategic_Event_Handler.cc

```diff
@@ -772,29 +773,16 @@ void HandleEarlyMorningEvents( void )
 			{
 				// ok, Devin's sector not loaded, so time to move!
 				// might be same sector as before, if so, oh well!
-				switch( Random( 5 ) )
+				uint8_t randomPlacement = Random(5);
```

Comment:
Make it `placement->sectorIds.size()` instead of 5 and there's both more freedom and the exception can be removed. Same with the others.